modules/javafx.swing/src/main/java/javafx/embed/swing/JFXPanel.java

```diff
@@ -810,6 +812,14 @@ protected void paintComponent(Graphics g) {
 
         Graphics gg = null;
         try {
+            ComponentOrientation cor = this.getComponentOrientation();
+            if (!cor.equals(ComponentOrientation.UNKNOWN)) {
+                String orient = cor.equals(ComponentOrientation.LEFT_TO_RIGHT)
```

Comment:
Why a String??  Should it be a simple boolean?

Reply:
Yes, Optimized...